daemon/api_notices.go

```diff
@@ -273,11 +275,15 @@ func noticeTypesViewableBySnap(types []state.NoticeType, r *http.Request) bool {
 		return false
 	}
 
+TYPE_LOOP:
 	for _, noticeType := range types {
 		allowedInterfaces := noticeReadInterfaces[noticeType]
-		if !strutil.ListContains(allowedInterfaces, iface) {
-			return false
+		for _, iface := range ifaces {
+			if strutil.ListContains(allowedInterfaces, iface) {
+				continue TYPE_LOOP
+			}
 		}
+		return false
```

Comment:
tiny nitpick, could we avoid the label jump, and use a simple flag and break?

```suggestion
		allowed := false
		for _, iface := range ifaces {
			if strutil.ListContains(allowedInterfaces, iface) {
				allowed = true
				break
			}
		}
		if !allowed {
			return false
		}
```

Reply:
I think I prefer the label jump here, but open to either if others agree with a more explicit flag.